### PR TITLE
TD-3034 Road Preview Component

### PR DESCRIPTION
**Pull request by jzhelyazkovtvm**
Closes [TD-3034](https://sce.myjetbrains.com/youtrack/issue/TD-3034/Add-new-RoadPreview-component-to-RUI)
Contributes to [TD-2983](https://sce.myjetbrains.com/youtrack/issue/TD-2983/Road-Hover-Preview)

## Changes

Created RoadPreview directory with RoadPreview stories file, test file, types file, index file.
The component is imported in main index.ts file.
Created mock data for rendering the component in storybook in RoadPreview.stories.
Created unit tests for the component in RoadPreview test file.
The current component uses another components for visualizing data, which are: UserAvatar, NoWrapTypography, TruncatedTooltip, VersionChip and LabelChipGroup.
2 Icons are imported in /static folder, for CarMaker logo and ASAM OpenSCENARIO.
The props createdAt, user and label are optional. All others are required

A brief description of what this pull request solves / introduces.

- New component named RoadPreview, which has to replace the current in 4 different places.

## Dependencies

N/A

## UI/UX

image

[Figma prototype](https://www.figma.com/design/0CVK945S1cs1ku0sUXcgv7/VIRTO.SCENE?node-id=10697-31067&node-type=section&t=WykPU7Yj4AA6K3Lf-0)

## Testing notes

We can view 3 testing scenarios in storybook from Figma prototype

## Author checklist before assigning a reviewer

- [x] Reviewed my own code-diff.
- [ ] ~Branch has been run in docker.~
- [x] PR assigned to me or an appropriate delegate.
- [x] Relevant labels added to the PR.
- [x] Appropriate tests have been added.
- [x] Lint and test workflows pass.

### CODE REVIEW

**Comment by m4manjesh** [comment]
@jzhelyazkovtvm Please fix the conflicts. I think this should be merging to a staging branch instead of main, please check with @lukemojo

**Review by m4manjesh — CHANGES_REQUESTED** [review]
Component looks good, some suggestions to improve the code. Next time please try to attach Figma if available. Also this PR has some changes of `DialogTitle` , may be because you created this branch from Main and i think this PRs base branch dont have those changes?

File: src/RoadPreview/RoadPreview.tsx [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+      default:
+        return CarMaker;
+    }
+  }
+
+  function hasOptionalProperty() {
+    return (label?.length && label?.length > 0) || createdAt || user;
+  }
+
+  function getCurrentDateFormatted() {
```
[the remark this answers is not in the document]
Comment (dmbarry86) [reply src/RoadPreview/RoadPreview.tsx:58]: I agree. I think each app should already be formatting the timstamps, for which we use a library.

**Comment by jzhelyazkovtvm** [comment]
> Component looks good, some suggestions to improve the code. Next time please try to attach Figma if available. Also this PR has some changes of `DialogTitle` , may be because you created this branch from Main and i think this PRs base branch dont have those changes?

After pushing changes I had conflicts and also the branch was created before the release from main branch. So then the base branch was changed to v8.0.0 which was the final release and I believe this is the reason I have changes on this branch.

File: src/RoadPreview/RoadPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    image,
+    description,
+    format,
+    formatVersion,
+    file,
+    createdAt,
+    user,
+    label
+  }: RoadPreviewProps,
+  key: React.Key
```
[the remark this answers is not in the document]
Comment (jzhelyazkovtvm) [reply src/RoadPreview/RoadPreview.tsx:38]: Removed

**Review by m4manjesh — CHANGES_REQUESTED** [review]
Please make the changes in the stories and add tests as discussed.

File: src/RoadPreview/RoadPreview.tsx [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+import { RoadPreviewProps } from "./RoadPreview.types";
+import TruncatedTooltip from "../TruncatedTooltip";
+import UserAvatar from "../UserAvatar/UserAvatar";
+import VersionChip from "../VersionChip/VersionChip";
+
+/**
+ * RoadPreview component for visualizing Road information about specific road
+ */
+
+export default function RoadPreview({
```
Comment (m4manjesh) [inline src/RoadPreview/RoadPreview.tsx:28]: we should change to using named exports as per the new style guide that we use in VIRTO.
https://github.com/IPG-Automotive-UK/virto/blob/qa/STYLEGUIDE.md#exports

```suggestion
export function RoadPreview({
```

File: src/index.ts [hunk 4]
```diff
@@ -29,6 +29,7 @@ export { default as CanvasItem, type CanvasItemProps } from "./CanvasItem";
 export { default as EditLabelDialog } from "./EditLabelDialog";
 export { default as RoadMarking, type RoadMarkingProps } from "./RoadMarking";
 export { default as RoadSurface, type RoadSurfaceProps } from "./RoadSurface";
+export { default as RoadPreview, type RoadPreviewProps } from "./RoadPreview";
```
Comment (m4manjesh) [inline src/index.ts:32]:
```suggestion
export { RoadPreview, type RoadPreviewProps } from "./RoadPreview";
```

File: src/RoadPreview/index.ts [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export { default } from "./RoadPreview";
```
Comment (m4manjesh) [inline src/RoadPreview/index.ts:1]:
```suggestion
export { RoadPreview } from "./RoadPreview";
```

File: src/RoadPreview/RoadPreview.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
+import { Meta, StoryFn } from "@storybook/react";
+
+import React from "react";
+import RoadPreview from "./RoadPreview";
```
Comment (m4manjesh) [inline src/RoadPreview/RoadPreview.stories.tsx:4]:
```suggestion
import { RoadPreview } from "./RoadPreview";
```

File: src/RoadPreview/RoadPreview.test.tsx [hunk 7]
```diff
@@ -0,0 +1,128 @@
+import { render, screen } from "@testing-library/react";
+
+import { Default } from "./RoadPreview.stories";
+import React from "react";
+import RoadPreview from ".";
```
Comment (m4manjesh) [inline src/RoadPreview/RoadPreview.test.tsx:5]:
```suggestion
import { RoadPreview } from ".";
```

**Review by m4manjesh — CHANGES_REQUESTED** [review]
Sorry i missed this my initial review. Please make the changes for default exports then we are good to go.

**Review by m4manjesh — APPROVED** [review]
LGTM

**Pull request merged**